jetbrains-core/src/software/aws/toolkits/jetbrains/services/sqs/toolwindow/PollMessagePane.kt

```diff
@@ -2,8 +2,105 @@
+import software.aws.toolkits.jetbrains.utils.ApplicationThreadPoolScope
+import software.aws.toolkits.resources.message
+import javax.swing.JLabel
 import javax.swing.JPanel
 
-class PollMessagePane {
+class PollMessagePane(
+    private val client: SqsClient,
+    private val queue: Queue
+) : CoroutineScope by ApplicationThreadPoolScope("PollMessagesPane") {
     lateinit var component: JPanel
+    lateinit var messagesAvailableLabel: JLabel
+    lateinit var tablePanel: SimpleToolWindowPanel
+    val messagesTable = MessagesTable()
+    var isSetup = false
+
+    private fun createUIComponents() {
+        tablePanel = SimpleToolWindowPanel(false, true)
+    }
+
+    init {
+        tablePanel.setContent(messagesTable.component)
+        addToolbar()
+    }
+
+    fun setUp() = launch {
+        isSetup = true
+        requestMessages()
+        addTotal()
+    }
+
+    suspend fun requestMessages() {
+        try {
+            withContext(Dispatchers.IO) {
+                val polledMessages: List<Message> = client.receiveMessage {
+                    it.queueUrl(queue.queueUrl)
+                    it.attributeNames(QueueAttributeName.ALL)
+                    it.maxNumberOfMessages(MAX_NUMBER_OF_MESSAGES)
+                }.messages()
+
+                polledMessages.forEach {
+                    messagesTable.tableModel.addRow(it)
+                }
+                messagesTable.setBusy(busy = false)
+            }
+        } catch (e: Exception) {
+            messagesTable.table.emptyText.text = message("sqs.failed_to_poll_messages")
+        }
+    }
+
+    suspend fun addTotal() {
+        try {
+            withContext(Dispatchers.IO) {
+                val numMessages = client.getQueueAttributes {
+                    it.queueUrl(queue.queueUrl)
+                    it.attributeNames(QueueAttributeName.APPROXIMATE_NUMBER_OF_MESSAGES)
+                }.attributes().getValue(QueueAttributeName.APPROXIMATE_NUMBER_OF_MESSAGES)
+
+                messagesAvailableLabel.text = message("sqs.messages.available.text") + numMessages
+            }
+        } catch (e: Exception) {
+            messagesAvailableLabel.text = message("sqs.failed_to_load_total")
+        }
+    }
+
+    private fun addToolbar() {
+        val actionGroup = DefaultActionGroup()
+        actionGroup.addAction(object : AnAction(message("general.refresh"), null, AllIcons.Actions.Refresh), DumbAware {
+            override fun actionPerformed(e: AnActionEvent) {
+                refreshTable()
+            }
+        })
+        tablePanel.toolbar = ActionManager.getInstance().createActionToolbar("PollMessagePane", actionGroup, false).component
+    }
+
+    // TODO: Add message table actions
+
+    private fun refreshTable() = launch {
+        messagesTable.setBusy(busy = true)
+        // Remove all entries
+        while (messagesTable.tableModel.rowCount != 0) {
+            messagesTable.tableModel.removeRow(0)
```

Comment:
nit: the table model can just be set to empty

Reply:
Please correct me if I'm wrong, but I think that's possible with `DefaultTableModel`, but not `ListTableModel`.